hoodie-client/src/main/java/com/uber/hoodie/HoodieReadClient.java

```diff
@@ -210,9 +210,10 @@ private void assertSqlContext() {
                 HoodieCommitMetadata metadata =
                         HoodieCommitMetadata.fromBytes(commitTimeline.getInstantDetails(commit).get());
                 // get files from each commit, and replace any previous versions
-                fileIdToFullPath.putAll(metadata.getFileIdAndFullPaths());
+                String basePath = hoodieTable.getMetaClient().getBasePath();
+                HashMap<String, String> pathsFromCurCommit = metadata.getFullPaths(basePath);
+                fileIdToFullPath.putAll(pathsFromCurCommit);
```

Comment:
nit: could this be just 1 line `fileIdToFullPath.putAll(metdata.getFullPaths(basePath))